src/main/java/org/sagebionetworks/bridge/workers/dynamodb/streams/StreamsUtils.java

```diff
@@ -0,0 +1,135 @@
+import org.sagebionetworks.bridge.dynamodb.DynamoUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.amazonaws.services.dynamodbv2.AmazonDynamoDB;
+import com.amazonaws.services.dynamodbv2.model.AttributeDefinition;
+import com.amazonaws.services.dynamodbv2.model.CreateTableRequest;
+import com.amazonaws.services.dynamodbv2.model.DescribeTableResult;
+import com.amazonaws.services.dynamodbv2.model.KeySchemaElement;
+import com.amazonaws.services.dynamodbv2.model.ProvisionedThroughput;
+import com.amazonaws.services.dynamodbv2.model.StreamSpecification;
+import com.amazonaws.services.dynamodbv2.model.StreamViewType;
+import com.amazonaws.services.dynamodbv2.model.TableDescription;
+import com.amazonaws.services.dynamodbv2.model.UpdateTableRequest;
+
+public final class StreamsUtils {
+
+    private static final Logger LOG = LoggerFactory.getLogger(StreamsUtils.class);
+
+    private static final Long MIN_READ_THROUGHPUT = 5L;
+
+    /**
+     * The list of all the existing tables for the current environment and user.
+     * This is a list of fully qualified table names.
+     */
+    public static List<String> getTables(final Config config, final AmazonDynamoDB dynamo) {
+        checkNotNull(config);
+        checkNotNull(dynamo);
+        final Map<String, TableDescription> tables = DynamoUtils.getExistingTables(config, dynamo);
+        return new ArrayList<String>(tables.keySet());
+    }
+
+    /**
+     * The list of tables that exist for the current environment and user. Only tables in the specified
+     * list are included if they indeed exist. This is a list of fully qualified table names.
+     */
+    public static List<String> getTables(final List<String> simpleTableNames, final Config config,
+            final AmazonDynamoDB dynamo) {
+        checkNotNull(simpleTableNames);
+        checkNotNull(config);
+        checkNotNull(dynamo);
+        final List<String> fqTableNames = simpleTableNames.stream()
+                .map(simpleTableName -> DynamoUtils.getFullyQualifiedTableName(simpleTableName, config))
+                .collect(Collectors.toList());
+        final Map<String, TableDescription> allTables = DynamoUtils.getExistingTables(config, dynamo);
+        return fqTableNames.stream()
+                .filter(fqTableName -> allTables.containsKey(fqTableName))
+                .collect(Collectors.toList());
+    }
+
+    public static List<DynamoStream> getStreams(final List<String> fqTableNames, final AmazonDynamoDB dynamo) {
+        checkNotNull(fqTableNames);
+        checkNotNull(dynamo);
+        final List<DynamoStream> streams = new ArrayList<>();
+        for (final String fqTableName : fqTableNames) {
+            if (!isStreamEnabled(fqTableName, dynamo)) {
+                LOG.info("Enabling stream for table " + fqTableName + "...");
+                enableStream(fqTableName, dynamo);
+                LOG.info("Stream enabled for table " + fqTableName + "... Done.");
+            }
```

Comment:
This is called during Spring Initialization. What happens if a new table is created while this is running? Will we have to restart our workers? Similarly with getReplicaTables().